From 8aeaa143a008ea7405103e372ec4ebcce3731180 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Thu, 25 Jul 2019 17:39:21 +0800
Subject: [PATCH] Add WRLinux1019 specific test case for rule
 set_password_hashing_algorithm_systemauth

Add WRLinux1019 specific particular test case for rule
 set_password_hashing_algorithm_systemauth.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
---
 .../bash/wrlinux1019.sh                            |  7 ++++++
 .../oval/wrlinux1019.xml                           | 25 ++++++++++++++++++++++
 2 files changed, 32 insertions(+)
 create mode 100644 linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/bash/wrlinux1019.sh
 create mode 100644 linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/oval/wrlinux1019.xml

diff --git a/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/bash/wrlinux1019.sh b/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/bash/wrlinux1019.sh
new file mode 100644
index 0000000..678ce1a
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/bash/wrlinux1019.sh
@@ -0,0 +1,7 @@
+# platform = multi_platform_wrlinux
+
+pamFile="/etc/pam.d/common-password"
+
+if ! grep -q "^password.*sufficient.*pam_unix.so.*sha512" $pamFile; then
+	sed -i --follow-symlinks "/^password.*sufficient.*pam_unix.so/ s/$/ sha512/" $pamFile
+fi
diff --git a/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/oval/wrlinux1019.xml b/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/oval/wrlinux1019.xml
new file mode 100644
index 0000000..3214412
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/oval/wrlinux1019.xml
@@ -0,0 +1,25 @@
+<def-group>
+  <definition class="compliance" id="set_password_hashing_algorithm_systemauth" version="1">
+    <metadata>
+      <title>Set Password Hashing Algorithm in /etc/pam.d/common-password</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The password hashing algorithm should be set correctly in /etc/pam.d/common-password.</description>
+    </metadata>
+    <criteria operator="AND">
+      <criterion test_ref="test_pam_unix_sha512" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="at_least_one_exists" comment="check /etc/pam.d/common-password for correct settings" id="test_pam_unix_sha512" version="1">
+    <ind:object object_ref="object_pam_unix_sha512" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object comment="check /etc/pam.d/common-password for correct settings" id="object_pam_unix_sha512" version="1">
+    <ind:filepath>/etc/pam.d/common-password</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*password[\s]+(?:(?:required)|(?:sufficient)|(?:\[.*default=ignore.*\]))[\s]+pam_unix\.so[\s]+.*sha512.*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+</def-group>
-- 
1.9.1

